src/cryptonote_core/pulse.cpp

```diff
@@ -444,6 +445,20 @@ void handle_messages_received_early_for(pulse_wait_stage &stage, void *quorumnet
   }
 }
 
+// Check the timestamps of the message queue and returns the average
+std::chrono::time_point<std::chrono::system_clock> average_timestamp(message_queue const &queue)
```

Comment:
This feels like it'd be better done as a median, malicious nodes can set a nonsensical timestamp (there's no penalty for this) and it'd cause unnecessary alarm to operators.

Median might be difficulty to come to consensus with the nodes in the quorum so you could bucket them to the closest 10s/5s etc.

Reply:
what about dropping outliers from the calculation entirely instead of bucketing?